src/scippneutron/tof/unwrap.py

```diff
@@ -333,7 +343,8 @@ def offset_from_wrapped(
     """
     time_bounds = frame_bounds['time']
     frame_period = frame_period.to(unit=elem_unit(time_bounds))
-    if time_bounds['bound', -1] - time_bounds['bound', 0] > frame_period:
+    diff = (time_bounds['bound', -1] - time_bounds['bound', 0]) - frame_period
+    if any(diff.flatten(to='x') > sc.scalar(0.0, unit=frame_period.unit)):
```

Comment:
I am confused by the `flatten`, why is it needed? Can we just
```suggestion
    if (diff > sc.scalar(0.0, unit=frame_period.unit)).any():
```

Reply:
The flatten was to handle both scalars and variables with dimensions in a generic way, but your suggestion also does this and avoids the flatten :+1: